bootstrap.py

```diff
@@ -149,15 +150,46 @@ def check_migration_version():
         sys.exit(1)
 
 
+def setup_yum_repo(url, gpg_key):
+    """
+    Configures a yum repository at /etc/yum.repos.d/katello-client-bootstrap-deps.repo
+    """
+    submanrepoconfig = SafeConfigParser()
+    submanrepoconfig.add_section('kt-bootstrap')
+    submanrepoconfig.set('kt-bootstrap', 'name', 'Subscription-manager dependencies for katello-client-bootstrap')
+    submanrepoconfig.set('kt-bootstrap', 'baseurl', url)
+    submanrepoconfig.set('kt-bootstrap', 'enabled', '1')
+    submanrepoconfig.set('kt-bootstrap', 'gpgcheck', '1')
+    submanrepoconfig.set('kt-bootstrap', 'gpgkey', gpg_key)
+    submanrepoconfig.write(open('/etc/yum.repos.d/katello-client-bootstrap-deps.repo', 'w'))
+    print_generic('Building yum metadata cache. This may take a few minutes')
+    exec_failexit('/usr/bin/yum makecache')
+
+
 def install_prereqs():
     """
     Install subscription manager and its prerequisites.
     """
-    print_generic("Installing subscription manager prerequisites")
+    print_generic("Checking subscription manager prerequisites")
+    if options.deps_repository_url:
+        print_generic("Enabling %s as a repository for dependency RPMs" % options.deps_repository_url)
+        setup_yum_repo(options.deps_repository_url, options.deps_repository_gpg_key)
+    yb = yumApi.YumBase()
+    yb.conf.cache = os.geteuid() != 1
+    pkg_list = yb.doPackageLists(patterns=['subscription-manager'])
     yum("remove", "subscription-manager-gnome")
-    yum("install", "subscription-manager 'subscription-manager-migration-*'")
+    if pkg_list.installed:
+        print_generic("subscription-manager is installed already")
```

Comment:
Actually we need it for both registration and migration cases. (As for older RHEL5 we need to grab product certs). Also, if we choose to add better support for checking for product certs before we register, we'd need it.